tsdb/ooo_head_read.go

```diff
@@ -243,7 +243,21 @@ func NewHeadAndOOOChunkReader(head *Head, mint, maxt int64, cr *headChunkReader,
 func (cr *HeadAndOOOChunkReader) ChunkOrIterable(meta chunks.Meta) (chunkenc.Chunk, chunkenc.Iterable, error) {
 	sid, _, isOOO := unpackHeadChunkRef(meta.Ref)
 	if !isOOO {
-		return cr.cr.ChunkOrIterable(meta)
+		chk, iter, err := cr.cr.ChunkOrIterable(meta)
+		switch {
+		case err != nil:
+			return chk, iter, err
+		case iter != nil:
+			// Assume that meta.Chunk is included. @krajorama: in-order head never returns an iterable so this is true by definition.
+			return chk, iter, nil
```

Comment:
Is `iter != nil` ever possible? Since `cr.cr` is always a `headChunkReader` that returns a chunk from `ChunkOrIterable()`. Not sure I understand the comment

Reply:
It's not possible as you say, but I'm not yet sure how to handle if eventually someone does return iterable. Maybe throw unexpected error?